File: Web/Engine/Entities/Light/AxLight.js
function AxLight()
{
    AxResource.call(this);

    this.typeId = AxLight.classId;
    this.resourceType = AxResourceType.Light;
       
    this.properties.Add(new AxProperty(new AxString('Color'), new AxVector4(1.0)));
    this.properties.Add(new AxProperty(new AxString('Intensity'), 1000.0));
}

AxLight.prototype.Dispose = function()
{
};

AxLight.prototype = Object.create(AxResource.prototype);


AxLight.classId = (AxResourceType.Light << 16) | 0;

AxLight.propertyIndex_Color     = AxResource.propertyIndex_ChildPropertiesIndex + 0;
AxLight.propertyIndex_Intensity = AxResource.propertyIndex_ChildPropertiesIndex + 1;

File: Engine/Axis/Source/Engine/Entities/Light/AxLight.cpp
//---------------------------------------------------------------------------------------------
//	Axis 3D Engine Copyright (C) 2017 Palitri
//
//	This software is released under the MIT License http://palitri.com/axis/license/
//
//	For help and documentation, visit http://palitri.com
//---------------------------------------------------------------------------------------------

#include "AxLight.h"

#include "..\..\Primitive\AxVector4.h"

AxLight::AxLight(void)
	: AxResource()
{
	this->typeId = AxLight::classId;
	this->resourceType = AxResourceType_Light;
       
    this->properties.Add(new AxProperty(AxString("Color"), AxVector4(1.0f)));
    this->properties.Add(new AxProperty(AxString("Intensity"), 1000.0f));
}

AxLight::~AxLight(void)
{
}
